[Antrea Multicluster] Antrea controller POD went in CrashLoopBackOff after enabling

**Describe the bug**
We have Antrea Scale ENV with Antrea-Multicluster feature enabled
Antrea v1.11.1

Antrea controller POD getting CrashLoopBackOff with enableStretchedNetworkPolicy as true in Antrea-controller config map section

k54 get po -A | grep antrea-controller
kube-system            antrea-controller-5948dd7cd9-69jf2                         0/1     CrashLoopBackOff   2 (4s ago)     28s


**To Reproduce**

> Enable parameter enableStretchedNetworkPolicy as true in cm at antrea-controller.conf section

  antrea-controller.conf: |
    featureGates:
      Traceflow: true
      AntreaPolicy: true
      NetworkPolicyStats: true
      Multicast: false
      Egress: true
      AntreaIPAM: false
      ServiceExternalIP: false
      SupportBundleCollection: false
      L7NetworkPolicy: false
      Multicluster: true
    tlsCipherSuites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384
    nodeIPAM: null
    multicluster:
      enableStretchedNetworkPolicy: true <<<<<<<<<<< Added this section


**Expected**

Antrea Controller POD should UP and RUNNING

**Actual behavior**

> Antrea Controller LOGS

> k55 logs antrea-controller-5948dd7cd9-69jf2 -n kube-system

E0817 08:32:14.359001       1 runtime.go:79] Observed a panic: runtime.boundsError{x:1, y:1, signed:true, code:0x0} (runtime error: index out of range [1] with length 1)
goroutine 342 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x30a97c0?, 0xc0000819f8})
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0006e6600?})
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/runtime/runtime.go:49 +0x75
panic({0x30a97c0, 0xc0000819f8})
	/usr/local/go/src/runtime/panic.go:884 +0x212
antrea.io/antrea/pkg/controller/labelidentity.constructMapFromLabelString({0xc0009d04f1, 0x6})
	/usr/src/antrea.io/antrea/pkg/controller/labelidentity/label_group_index.go:130 +0x132
antrea.io/antrea/pkg/controller/labelidentity.newLabelIdentityMatch({0xc0009d0460?, 0x97?}, 0x8)
	/usr/src/antrea.io/antrea/pkg/controller/labelidentity/label_group_index.go:139 +0xa5
antrea.io/antrea/pkg/controller/labelidentity.(*LabelIdentityIndex).AddLabelIdentity(0xc0006e6600, {0xc0009d0460, 0x97}, 0x8)
	/usr/src/antrea.io/antrea/pkg/controller/labelidentity/label_group_index.go:381 +0x1f6
antrea.io/antrea/pkg/controller/labelidentity.(*Controller).addLabelIdentity(0xc0008ae420, {0x31f1060?, 0xc000b99560})
	/usr/src/antrea.io/antrea/pkg/controller/labelidentity/controller.go:113 +0x199
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/tmp/gopath/pkg/mod/k8s.io/client-go@v0.24.9/tools/cache/controller.go:232
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/tmp/gopath/pkg/mod/k8s.io/client-go@v0.24.9/tools/cache/shared_informer.go:818 +0x134
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00071bf38?, {0x376fd40, 0xc000d04120}, 0x1, 0xc00071e060)
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc001318100?)
	/tmp/gopath/pkg/mod/k8s.io/client-go@v0.24.9/tools/cache/shared_informer.go:812 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/tmp/gopath/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:71 +0x85
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 342 [running]:
 

**Versions:**
Antrea v1.11.1

**Additional context**
<!-- Add any other context about the problem here, such as Antrea logs, kubelet logs, etc. -->

<!-- (Please consider pasting long output into a [GitHub gist](https://gist.github.com) or any other pastebin.) -->

--- a/multicluster/controllers/multicluster/member/labelidentity_controller.go
+++ b/multicluster/controllers/multicluster/member/labelidentity_controller.go
@@ -326,7 +326,7 @@ func GetNormalizedLabel(nsLabels, podLabels map[string]string, ns string) string
 		// label is guaranteed to have Namespace name information.
 		nsLabels[v1.LabelMetadataName] = ns
 	}
-	return "ns:" + labels.FormatLabels(nsLabels) + "&pod:" + labels.FormatLabels(podLabels)
+	return "ns:" + labels.Set(nsLabels).String() + "&pod:" + labels.Set(podLabels).String()
 }
 
 // getResourceExportNameForLabelIdentity retrieves the ResourceExport name for exporting

--- a/multicluster/controllers/multicluster/member/labelidentity_controller_test.go
+++ b/multicluster/controllers/multicluster/member/labelidentity_controller_test.go
@@ -271,6 +271,13 @@ func TestGetNormalizedLabel(t *testing.T) {
 			map[string]string{"region": "west"},
 			"ns:kubernetes.io/metadata.name=test-ns,region=west&pod:purpose=test",
 		},
+		{
+			"no Pod label",
+			"test-ns",
+			map[string]string{},
+			map[string]string{"region": "west"},
+			"ns:kubernetes.io/metadata.name=test-ns,region=west&pod:",
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

--- a/pkg/controller/labelidentity/label_group_index.go
+++ b/pkg/controller/labelidentity/label_group_index.go
@@ -124,6 +124,9 @@ func (l *labelIdentityMatch) matches(s *selectorItem) bool {
 // constructMapFromLabelString parses label string of format "app=client,env=dev" into a map.
 func constructMapFromLabelString(s string) map[string]string {
 	m := map[string]string{}
+	if s == "" {
+		return m
+	}
 	kvs := strings.Split(s, ",")
 	for _, kv := range kvs {
 		kvpair := strings.Split(kv, "=")

--- a/pkg/controller/labelidentity/label_group_index_test.go
+++ b/pkg/controller/labelidentity/label_group_index_test.go
@@ -42,6 +42,8 @@ var (
 	selectorC     = types.NewGroupSelector("", pSelDB, nil, nil, nil)
 	selectorD     = types.NewGroupSelector("testing", pSelDB, nil, nil, nil)
 	selectorE     = types.NewGroupSelector("random", pSelDB, nil, nil, nil)
+	selectorF     = types.NewGroupSelector("", nil, nsSelTest, nil, nil)
+	selectorG     = types.NewGroupSelector("testing", nil, nil, nil, nil)
 	selectorItemA = &selectorItem{
 		selector: selectorA,
 	}
@@ -57,9 +59,16 @@ var (
 	selectorItemE = &selectorItem{
 		selector: selectorE,
 	}
+	selectorItemF = &selectorItem{
+		selector: selectorF,
+	}
+	selectorItemG = &selectorItem{
+		selector: selectorG,
+	}
 	labelA = "ns:kubernetes.io/metadata.name=testing,purpose=test&pod:app=web"
 	labelB = "ns:kubernetes.io/metadata.name=testing,purpose=test&pod:app=db"
 	labelC = "ns:kubernetes.io/metadata.name=nomatch,purpose=nomatch&pod:app=db"
+	labelD = "ns:kubernetes.io/metadata.name=testing,purpose=test&pod:"
 )
 
 func TestLabelIdentityMatch(t *testing.T) {
@@ -127,6 +136,41 @@ func TestLabelIdentityMatch(t *testing.T) {
 			label:       labelC,
 			selector:    selectorItemE,
 			expectMatch: false,
+		},
+		{
+			label:       labelD,
+			selector:    selectorItemA,
+			expectMatch: false,
+		},
+		{
+			label:       labelD,
+			selector:    selectorItemB,
+			expectMatch: false,
+		},
+		{
+			label:       labelD,
+			selector:    selectorItemD,
+			expectMatch: false,
+		},
+		{
+			label:       labelD,
+			selector:    selectorItemF,
+			expectMatch: true,
+		},
+		{
+			label:       labelA,
+			selector:    selectorItemG,
+			expectMatch: true,
+		},
+		{
+			label:       labelC,
+			selector:    selectorItemG,
+			expectMatch: false,
+		},
+		{
+			label:       labelD,
+			selector:    selectorItemG,
+			expectMatch: true,
 		},
 	}
 	for _, tt := range tests {

